From a7c58f6ec716487176debad1762116cf62b5bd77 Mon Sep 17 00:00:00 2001
From: Raymond Tan <raymond.tan@intel.com>
Date: Tue, 16 Jun 2020 14:44:38 +0800
Subject: [PATCH] i915: EHL specific Wa_14010685332

It was found out that the EHL behaves differently from ICL
with respect to ICL, which EHL adopted from.

The real WA is being investigated, this patch will be carried
temporarily in meta layer meanwhile, that has been tested by several
setups.

Signed-off-by: Raymond Tan <raymond.tan@intel.com>
---
 drivers/gpu/drm/i915/i915_irq.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/gpu/drm/i915/i915_irq.c b/drivers/gpu/drm/i915/i915_irq.c
index f37155c..f5d7fe1 100644
--- a/drivers/gpu/drm/i915/i915_irq.c
+++ b/drivers/gpu/drm/i915/i915_irq.c
@@ -2862,6 +2862,10 @@ static void gen11_irq_reset(struct drm_i915_private *dev_priv)
 				 SBCLK_RUN_REFCLK_DIS, SBCLK_RUN_REFCLK_DIS);
 		intel_uncore_rmw(uncore, SOUTH_CHICKEN1,
 				 SBCLK_RUN_REFCLK_DIS, 0);
+	} else if (INTEL_PCH_TYPE(dev_priv) == PCH_MCC) {
+		printk(KERN_ERR "Setting south_chicken1 SBCLK_RUN_REFCLK_DIS\n");
+		intel_uncore_rmw(uncore, SOUTH_CHICKEN1,
+				 SBCLK_RUN_REFCLK_DIS, SBCLK_RUN_REFCLK_DIS);
 	}
 }
 
-- 
2.7.4

